### ISSUE TITLE
A swap request gets deleted but no tokens returned until another request is made

### ISSUE THREAD

**Issue by konnov** [tendermint/liquidity#425]
## Summary of Bug

A swap request is processed and is shown as `to_be_deleted: true`. The tokens are not returned, unless another deposit or swap request is made.

Surfaced from @informalsystems audit of v1.2.6-rc1

## Version

v1.2.6-rc1 of @tendermint/liquidity, commit fa647b0fefe5508e9c975b3d4f095db2d3d20a13 of @b-harvest/gravity-dex.

## Steps to Reproduce

Set `app_state["liquidity"]["params"]["unit_batch_height"]` to 10.

Initialize user accounts `user1` and `user2`:

```
gaiad add-genesis-account $(gaiad keys show user1 --keyring-backend test -a) 10000000000stake,10000000000uatom,500000000000uusd
gaiad add-genesis-account $(gaiad keys show user2 --keyring-backend test -a) 10000000000stake,10000000000uatom,500000000000uusd
```

Perform the following commands after starting `gaiad`:

```
$ gaiad tx liquidity create-pool 1 1000001uatom,1000001uusd --from=user2 --chain-id=testing \
  --keyring-backend=test --yes -b block
$ gaiad tx liquidity swap 1 1 100uusd uatom 1 0.003000000000000000 --from=user1 --chain-id=testing \
  --keyring-backend=test --yes -b block
```

After 10 blocks, the status of the swap request looks as follows:

```
gaiad query liquidity swaps 1
pagination:
  next_key: null
  total: "1"
swaps:
- exchanged_offer_coin:
    amount: "0"
    denom: uusd
  executed: true
  msg:
    demand_coin_denom: uatom
    offer_coin:
      amount: "100"
      denom: uusd
    offer_coin_fee:
      amount: "0"
      denom: uusd
    order_price: "1.000000000000000000"
    pool_id: "1"
    swap_requester_address: cosmos1zwq5j5m8ln9t59sjqgrjslvacqn2attpv3v0gq
    swap_type_id: 1
  msg_height: "4"
  msg_index: "1"
  order_expiry_height: "4"
  remaining_offer_coin:
    amount: "100"
    denom: uusd
  reserved_offer_coin_fee:
    amount: "0"
    denom: uusd
  succeeded: false
  to_be_deleted: true
```

Still, `user1` does not get their tokens refunded, e.g., after 50 blocks:

```
gaiad query bank balances cosmos1zwq5j5m8ln9t59sjqgrjslvacqn2attpv3v0gq
balances:
- amount: "10000000000"
  denom: stake
- amount: "10000000000"
  denom: uatom
- amount: "499999999900"
  denom: uusd
pagination:
  next_key: null
  total: "0"
```

If another swap request or a deposit is processed later, the tokens are refunded.

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by dongsam** [comment]
Hi @konnov, Thank you for opening the issue.
There was a bug about `order_expiry_height ` and `params.UnitBatchHeight` logics on [v1.2.6-rc1](https://github.com/tendermint/liquidity/releases/tag/v1.2.6-rc1), 
and the bug is resolved on #372 PR, from [v1.2.6-rc2](https://github.com/tendermint/liquidity/releases/tag/v1.2.6-rc2)

Fortunately, based on the current latest version of liquidity [v1.2.8](https://github.com/tendermint/liquidity/releases/tag/v1.2.8), and Gaia app using the version [gravity-dex v1.0.4](https://github.com/b-harvest/gravity-dex/releases/tag/v1.0.4),  it was confirmed that it was not reproduced.

**Comment by konnov** [comment]
Thanks! It works with the version v1.2.8. So I will close the issue.
